src/components/Modals/PublishWizardCollectionModal/ConfirmCollectionItemsStep/ConfirmCollectionItemsStep.tsx

```diff
@@ -0,0 +1,106 @@
+import React from 'react'
+import { Button, Column, Loader, Mana, Modal, Row, Table } from 'decentraland-ui'
+import { Props } from '../PublishWizardCollectionModal.types'
+import { t } from 'decentraland-dapps/dist/modules/translation/utils'
+import CollectionProvider from 'components/CollectionProvider'
+import { Item } from 'modules/item/types'
+
+import './ConfirmCollectionItemsStep.css'
+import ItemImage from 'components/ItemImage'
+import ItemBadge from 'components/ItemBadge'
+import RarityBadge from 'components/RarityBadge'
+import { isFree } from 'modules/item/utils'
+import { Network } from '@dcl/schemas'
+import { ethers } from 'ethers'
+
+const HUGE_PAGE_SIZE = 20
```

Comment:
If we stay with the CollectionProvider, we should increase this to 50 I think, otherwise some items wouldn't be shown if the collection has >20, right?